core/include/log.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  core_util
+ * @{
+ *
+ * @file        log.h
+ * @brief       System logging header
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef __LOG_H
+#define __LOG_H
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+enum {
+    LOG_NONE,
+    LOG_ERROR,
+    LOG_WARNING,
+    LOG_INFO,
+    LOG_DEBUG,
+    LOG_ALL
+};
+
+#ifndef LOG_LEVEL
+#define LOG_LEVEL LOG_ALL
+#endif
+
+/**
+ * @def Log message if level <= LOG_LEVEL and
+ */
+
+#define log(level, ...) if (level <= LOG_LEVEL) _log(level, __VA_ARGS__)
```

Comment:
I guess its a matter of taste, but I think I would prefer to capitalize the log macros `LOG_ERROR()` etc, to make clear that these trigger these are pre-processor functions...